near-contract-standards/src/access/mod.rs

```diff
@@ -1,129 +1,281 @@
+use near_sdk::serde::{Deserialize, Serialize};
 use near_sdk::{env, require, AccountId};
-use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 
+pub type RoleId = [u8; 32];
+
+/// Contract module which provides a basic access control mechanism, where
+/// there is an account (an [`Self::owner`]) that can be granted exclusive access to
+/// specific functions.
+///
+/// By default, the owner account will be the [one that deploys](env::predecessor_account_id())
+/// the contract. This can later be changed with [`Self::transfer_ownership()`].
+///
+/// This module makes available the modifier [`Self::only_owner()`], which can be applied to your
+///  functions to restrict their use to the owner.
 #[derive(BorshSerialize, BorshDeserialize)]
 pub struct Ownable {
-    pub owner: AccountId,
-    pub default_owner: AccountId,
+    pub owner: Option<AccountId>,
 }
 
 impl Ownable {
+    /// Initializes the contract setting
+    /// the [`predecessor account`](env::predecessor_account_id())
+    /// as the initial [`Self::owner`].
     pub fn new() -> Self {
-        Self {
-            owner: env::predecessor_account_id(),
-            default_owner: "0000000000".parse::<AccountId>().unwrap(),
-        }
+        Self { owner: Some(env::predecessor_account_id()) }
     }
 
-    pub fn owner(&self) -> AccountId {
+    /// Returns the address of the current [`Self::owner`].  
+    /// Panics if it has no owner.
+    pub fn owner(&self) -> Option<AccountId> {
         self.owner.clone()
     }
 
+    /// Has no effect if called by the owner.
+    /// Panics otherwise.
     pub fn only_owner(&self) {
-        require!(env::predecessor_account_id() == self.owner(), "Ownable: caller is not the owner");
+        require!(
+            Some(env::predecessor_account_id()) == self.owner,
+            "Ownable: caller is not the owner"
+        );
     }
 
-    pub fn renounce_ownership(&mut self) {
+    /// Permanently leaves the contract without an [`Self::owner`].  
+    /// Can only be called by the current owner.
+    ///
+    /// # WARNING
+    ///
+    /// Renouncing ownership will leave the contract without an owner,
+    /// thereby removing any functionality that is only available to the owner.  
+    /// ie. future calls into [`Self::only_owner()`] will always panic.
+    pub fn permanently_renounce_ownership(&mut self) {
         self.only_owner();
-        self.owner = self.default_owner.clone();
+        self.owner = None;
     }
 
+    /// Tranfers [ownership](Self::owner) of the contract to a new account `new_owner`.  
+    /// Can only be called by the current owner.
     pub fn transfer_ownership(&mut self, new_owner: AccountId) {
         self.only_owner();
-        require!(new_owner != self.default_owner, "Ownable: new owner is undefined");
-        self.owner = new_owner;
+        self.owner = Some(new_owner);
     }
 }
 
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+#[serde(crate = "near_sdk::serde")]
```

Comment:
Haha, I don't know what to believe anymore. I took example from their AccountId, where they use the original serde library. See https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/types/account_id.rs#L2

Now that I'm looking more, they use the near version of serde in FungibleToken. See https://github.com/near/near-sdk-rs/blob/master/near-contract-standards/src/fungible_token/metadata.rs#L9

I think there is literally no difference between them, but since the original `serde` requires an external dependencies, I will go for the near version.

Reply:
yup, the sdk is re-exporting serde. I also think that if it's already accessible, it's not needed to import it directly.  
Idk if there is another reason besides that. Probably because.. if they wanted to update serde version (if they are using specific versioning for serde), they would have to change in dozens of places, but if they use the sdk one, they just need to change in the sdk